syntax = "proto3";

message MenuItem {
  string id = 1;
  string name = 2;
  int32 quantity = 3;
  float price = 4;
}

message Order {
  string id = 1;
  repeated MenuItem items = 2;
}

message OrderList {
  repeated Order orders = 1;
}

message OrderId {
  string id = 1;
}

message Empty {}

service OrderService {
  rpc AddOrder(Order) returns (Order) {}
  rpc GetAllOrders(Empty) returns (OrderList) {}
  rpc GetOrder(OrderId) returns (Order) {}
  rpc UpdateOrder(Order) returns (Order) {}
  rpc DeleteOrder(OrderId) returns (Empty) {}
}
